### PR TITLE
fix: Failed to execute 'encode' on 'TextEncoder': parameter 1 is not of type 'String' in Edge Runtime SSR

### PR DESCRIPTION
**Pull request by AirBorne04**
## Changes

- fix small issue in SSR mode #5915
- this is not fixing the Solid build for vercel, so for Solid it will still state something like `Error: Pa is not supported in the browser, returning undefined` that is a vercel adapter issue only and we need to update the build steps there.

## Testing

Tested example applications for react and solid on vercel.

### CODE REVIEW

**Comment by AirBorne04** [comment]
These fixes are only for the issues in core / to fix eg cloudflare for now. The solid + vercel needs an additional fix in the vercel adapter.

**Review by bholmesdev — APPROVED** [review]
NWTWWHB (not worse than what we had before!)

File: packages/astro/src/runtime/server/render/component.ts [hunk 1]
```diff
@@ -261,7 +261,7 @@ If you're still stuck, please open an issue on GitHub or join us at https://astr
 
 			if (isPage || renderer?.name === 'astro:jsx') {
 				yield html;
-			} else {
+			} else if (html) {
```
Comment (bholmesdev) [inline packages/astro/src/runtime/server/render/component.ts:264]: Could this cause the server to hang if there is _not_ HTML? Could we have an `else` case with a detailed internal error, or does fall-through work?

**Review by bholmesdev — COMMENTED** [review]
Changing my review sorry! Comment before merging

File: packages/astro/src/runtime/server/render/component.ts (hunk 1 above)
Comment (AirBorne04) [reply packages/astro/src/runtime/server/render/component.ts]: yeah you are right i replace it with an empty string return now, that should do.

**Review by bholmesdev — APPROVED** [review]

**Pull request merged**